wagon.py

```diff
@@ -0,0 +1,1014 @@
+########
+# Copyright (c) 2014 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+import os
+import re
+import sys
+import time
+import json
+import shutil
+import urllib
+import urllib2
+import tarfile
+import zipfile
+import logging
+import platform
+import tempfile
+import subprocess
+import pkg_resources
+from threading import Thread
+from contextlib import closing
+
+import click
+try:
+    import virtualenv
+except ImportError:
+    pass
+from wheel import pep425tags as wheel_tags
+
+
+REQUIREMENT_FILE_NAMES = ['dev-requirements.txt', 'requirements.txt']
+METADATA_FILE_NAME = 'package.json'
+DEFAULT_WHEELS_PATH = 'wheels'
+
+DEFAULT_INDEX_SOURCE_URL = 'https://pypi.python.org/pypi/{0}/json'
```

Comment:
this is not the final value so the name `FORMAT` or `TEMPLATE` should probably be appended